stl/inc/print

```diff
@@ -0,0 +1,225 @@
+        __std_get_unicode_console_handle_from_file_stream(_File_stream)};
+
+    // See the documentation for __std_unicode_console_retrieval_result to understand why we do this.
+    bool _Is_unicode_console;
+
+#pragma warning(push)
+#pragma warning(disable : 4061) // enumerator not explicitly handled by switch label
+    switch (_Unicode_console_retrieval_result._Error) {
+    case __std_win_error::_Success:
+        {
+            _Is_unicode_console = true;
+            break;
+        }
+
+    case __std_win_error::_File_not_found:
+        {
+            _Is_unicode_console = false;
+            break;
+        }
+
+    case __std_win_error::_Not_supported:
+        [[unlikely]] return;
+
+    default:
+        [[unlikely]] _STD _Throw_system_error_from_std_win_error(_Unicode_console_retrieval_result._Error);
+    }
+#pragma warning(pop)
+
+    if (_Is_unicode_console) {
+        const bool _Was_flush_successful = (_STD fflush(_Stream) == 0);
+
+        if (!_Was_flush_successful) [[unlikely]] {
+            throw system_error{_STD make_error_code(static_cast<errc>(errno))};
+        }
+
+        const __std_win_error _Console_print_result =
+            __std_print_to_unicode_console(_Unicode_console_retrieval_result._Console_handle, _Str.data(), _Str.size());
+
+        if (_Console_print_result != __std_win_error::_Success) [[unlikely]] {
+            _STD _Throw_system_error_from_std_win_error(_Console_print_result);
+        }
+    } else {
+        _STD _Print_noformat_nonunicode(_Stream, _Str);
+    }
+}
+
+template <bool _Add_newline>
+void _Vprint_unicode_impl(FILE* const _Stream, const string_view _Fmt_str, const format_args _Fmt_args) {
+    const string _Output_str = [&]() {
+        if constexpr (_Add_newline) {
+            return _STD vformat(_Fmt_str, _Fmt_args) + '\n';
+        } else {
+            return _STD vformat(_Fmt_str, _Fmt_args);
+        }
+    }();
+
+    _Print_noformat_unicode(_Stream, _Output_str);
+}
+
+template <bool _Add_newline, class... _Types>
+void _Print_impl(FILE* const _Stream, const format_string<_Types...> _Fmt, _Types&&... _Args) {
+    constexpr bool _Has_format_args       = (sizeof...(_Types) > 0);
+    constexpr auto _Escape_braces_closure = [](const string_view _Old_str) {
+        if (_Old_str.empty()) {
+            return string{};
+        }
+
+        string _Escaped_str;
+        size_t _Escaped_str_expected_size = _Old_str.size();
+
+        if constexpr (_Add_newline) {
+            ++_Escaped_str_expected_size;
+        }
+
+        _Escaped_str.resize_and_overwrite(_Escaped_str_expected_size, [_Old_str](char* _Dest_ptr, const size_t) {
+            char _Prev_char           = _Old_str.front();
+            size_t _Num_chars_written = 1;
+            *_Dest_ptr++              = _Prev_char;
+
+            for (const auto _Curr_char : _Old_str.substr(1)) {
+                if ((_Curr_char != '{' || _Prev_char != '{') && (_Curr_char != '}' || _Prev_char != '}')) {

```

Comment:
What's the output if the string contains `{{{{`?

Reply:
I realized this bug existed just last night. As you imagine, the output for such a string was incorrect (`{` instead of `{{` in your example). I did push a commit to fix this, though.